apps/notifications/src/panel/templates/button-trash.jsx

```diff
@@ -3,32 +3,36 @@
  */
 import PropTypes from 'prop-types';
 
-import React from 'react';
+import React, { useContext } from 'react';
 import { localize } from 'i18n-calypso';
+import { connect } from 'react-redux';
 
 /**
  * Internal dependencies
  */
-import { trashNote } from '../flux/note-actions';
+import { trashNote } from '../state/notes/thunks';
 import ActionButton from './action-button';
 import { keys } from '../helpers/input';
+import { RestClientContext } from '../Notifications';
 
-const TrashButton = ( { note, translate } ) => (
-	<ActionButton
-		{ ...{
-			icon: 'trash',
-			isActive: false,
-			hotkey: keys.KEY_T,
-			onToggle: () => trashNote( note ),
-			text: translate( 'Trash', { context: 'verb: imperative' } ),
-			title: translate( 'Trash comment', { context: 'verb: imperative' } ),
-		} }
-	/>
-);
+const TrashButton = ( { note, translate, trashNote } ) => {
+	const restClient = useContext( RestClientContext );
+
+	return (
+		<ActionButton
+			icon={ 'trash' }
```

Comment:
Nit: I'd alter that to:
```suggestion
			icon="trash"
```

because it's not only more common but also more readable IMHO. Also, 4 less bytes of code! 💪 😉